File: Linked_list_insertion.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None

class LinkedList:
    def __init__(self):
        self.head=None

    def insert_begin(self,val):
        newnode=Node(val)
        newnode.next=self.head
        self.head=newnode

    def insert_end(self,val):
        newnode=Node(val)
        if not self.head:self.head=newnode;return
        temp=self.head
        while temp.next:temp=temp.next
        temp.next=newnode

    def insert_pos(self,val,pos):
        if pos==0: self.insert_begin(val);return
        temp=self.head
        for _ in range(pos-1):
            if not temp:return
            temp=temp.next
        if not temp:return
        newnode=Node(val)
        newnode.next=temp.next
        temp.next=newnode

    def display(self):
        temp=self.head
        while temp:print(temp.data,end=" ");temp=temp.next
        print()

ll=LinkedList()
ll.insert_end(10);ll.insert_begin(5);ll.insert_end(20);ll.insert_pos(15,2)
ll.display()

File: Linked_list_insertion.c
#include <stdio.h>
#include <stdlib.h>

struct Node{
    int data;
    struct Node* next;
};

struct Node* head=NULL;

void insert_begin(int val){
    struct Node* newnode=malloc(sizeof(struct Node));
    newnode->data=val;
    newnode->next=head;
    head=newnode;
}

void insert_end(int val){
    struct Node* newnode=malloc(sizeof(struct Node));
    newnode->data=val;
    newnode->next=NULL;
    if(!head){head=newnode;return;}
    struct Node* temp=head;
    while(temp->next)temp=temp->next;
    temp->next=newnode;
}

void insert_pos(int val,int pos){
    if(pos==0){insert_begin(val);return;}
    struct Node* temp=head;
    for(int i=0;i<pos-1 && temp;i++)temp=temp->next;
    if(!temp)return;
    struct Node* newnode=malloc(sizeof(struct Node));
    newnode->data=val;
    newnode->next=temp->next;
    temp->next=newnode;
}

void display(){
    struct Node* temp=head;
    while(temp){printf("%d ",temp->data);temp=temp->next;}
    printf("\n");
}

int main(){
    insert_end(10);insert_begin(5);insert_end(20);insert_pos(15,2);
    display();
    return 0;
}
